EzMap.Api/Controllers/PoiCollectionController.cs

```diff
@@ -16,13 +17,40 @@ public class PoiCollectionController : ControllerBase
     [HttpPost("")]
     public async Task<IActionResult> AddPoiCollection([FromBody] PoiCollectionCreateDto dto,
         [FromServices] IUnitOfWork uow,
-        [FromServices] IIdentityService identityService)
+        [FromServices] IIdentityService identityService, [FromServices] IElasticSearchService elasticSearchService)
     {
-        uow.PoiCollectionRepository.AddPoiCollection(dto.WithUserId(identityService.GetUserId()));
-
-        return await uow.SaveAsync() > 0
-            ? Ok("Your poi collection is created successfully!")
-            : new StatusCodeResult(StatusCodes.Status500InternalServerError);
+        try
+        {
+            var poiCollectionId = uow.PoiCollectionRepository.AddPoiCollection(dto.WithUserId(identityService.GetUserId()));
+            var dbResult = await uow.SaveAsync();
+            if (dbResult <= 0)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+            
+            var poiCollectionCreateIndex = new PoiCollectionCreateIndex(poiCollectionId, dto.Name, dto.Description);
```

Comment:
Why is the model for indexing an entity called Index? Is it an index or is it a model to index the entity?

Reply:
The purpose of this is treat it as a model for indexing an entity